research/workflow.py

```
@@ -0,0 +1,110 @@
+      size=size,
+      state=DataItemState.READY,
+      producer=None
+    )
+    data_item_id = len(self.data_items)
+    self.data_items.append(data_item)
+    self.inputs.add(data_item_id)
+    self.outputs.add(data_item_id)
+    return data_item_id
+
+  def update_task_state(self, task_id: int, state: TaskState):
+    task = self.tasks[task_id];
```

Ruff (0.15.12)

[error] Statement ends with an unnecessary semicolon

Remove unnecessary semicolon

(E703)